elixir/apps/web/lib/web/live/gateways_live/edit.ex

```diff
@@ -9,17 +16,17 @@ defmodule Web.GatewaysLive.Edit do
           %{label: "Home", path: ~p"/#{@subject.account}/dashboard"},
           %{label: "Gateways", path: ~p"/#{@subject.account}/gateways"},
           %{
-            label: "gcp-primary",
-            path: ~p"/#{@subject.account}/gateways/DF43E951-7DFB-4921-8F7F-BF0F8D31FA89"
+            label: "#{@gateway.name_suffix}",
```

Comment:
Our names should be built from prefix+suffix:
```suggestion
            label: "#{@gateway.group.name_prefix}-#{@gateway.name_suffix}",
```
but maybe we should add a struct helper to `gateways/gateway.ex`:
```elixir
def name(gateway) do
  "#{gateway.group.name_prefix}-#{gateway.name_suffix}"
end
```

Reply:
Yeah, I actually had that originally, but then when we talked about changing the naming of Gateways/Gateway Groups/Gateway Instances/etc... I figured I would leave this as is for now until we can settle on what things are going to be called.

Along with that, I was wondering what we actually wanted to allow the user to change and not change.  By that I mean, allowing the user to change (what we currently call) a Gateway Group makes sense, but that should then probably be called out specifically, e.g. `Edit Gateway Group` (again, just calling it group for now).  But I was curious as to whether or not we wanted to allow people to change the actual Gateway instance names from the UI or should that only be allowed to happen from the config on the Gateway Instance itself?  (or another question, should we even allow them to change the name at all for an instance?)

So, all of that said, I'm happy to update this to display whatever at this point, but was wondering if we wanted to just let this stay as is for now since we'll probably be changing terminology?

Reply:
@bmanifold The name will consist of two components anyways, no matter how we rename it so I feel that it's better to show the full name right away.

I think the user can't change the `gateway` right now, only `name_prefix` and `tags` of a `gateway_group`. `name_suffix` is the only technically "editable" field on the `gateway` which should be autogenerated by the gateway when it connects.

Reply:
Yeah, I can see what you're saying, however, this is where I feel like getting our terminology fixed first might be a good thing, because in the case you just described, what we would be able to edit would be the `gateway group`, which means there wouldn't really be any need to display the `name_suffix` on this page.

Along with that, we wouldn't really need an edit page at all for an individual `gateway`, since there wouldn't be any editable fields, only for the `gateway group`, which would should then have a different title for the whole page I would think.

Reply:
@bmanifold You are right. I think we should discuss what we render on that list page. There are a few options:

1. We can show the list of gateways grouped by the "group" (a vertical line with a wide colspan and buttons to edit the group). Something similar to this GitHub Projects view (sorry I'm really bad at drawing :): 
<img width="1722" alt="Screenshot 2023-07-12 at 11 49 26" src="https://github.com/firezone/firezone/assets/1877644/468a72c6-1ae4-41a4-aef7-25a483cf4bb9">

This one is my favorite because we still can show things like status, version, ips for individual gateways while transparently showing that they are grouped.

2. We can show just groups with a count of gateways per group. This means we will need a separate page for gateways in a group.

3. Or just gateways with a group name as a separate column.

cc @jamilbk

Reply:
So I think option (1) makes the most sense long-term, but if it's complicated to implemented we can stick with option (3) for now due to speed of development. It's an easy thing to add rollups for later. My thinking is there are lots of other bottlenecks in the product an organization will hit before they need more than a couple gateway groups. E.g. Windows client.

Reply:
I started working on trying to do option (1) this morning and attached is what I came up with.  I think long term it will work well, but this was only a static template.  I think getting this in to a component is going to be some effort and I'm wondering if that effort is better spent just getting all of the views showing live data.

Thoughts?

<img width="1520" alt="Screenshot 2023-07-13 at 10 50 03 AM" src="https://github.com/firezone/firezone/assets/2646332/98990506-9785-457f-914a-a9273bf04b40">

Reply:
@bmanifold Nice work man, I think this works for now. I'm not sure if we'll have any other areas that need colspans like this yet, so maybe we can kick the "extract into a component" can the road.

Reply:
Sorry, I don't think I was clear, this is just static data.  If we want to use this now I'd still need to do some work to get live data in to this and I think that would essentially be putting this in to a component.

Reply:
This looks conceptually good. I think it's worth to put it into a component now (and have you practice building one at the same time), because that will make you feel more comfortable reworking other pages and we won't need to go back and revisit this